### DISCUSSION TITLE
I'm new and confused lol

### DISCUSSION THREAD

**Discussion by jamie105** [CustomiesDevs/Customies#43] [Q&A] [answered]
Ello I'm very new to pocketmine development and I'm just confused af...

So in my main.php I have under OnEnable()

CustomiesItemFactory::getInstance()->registerItem(Item::class, "customies:test", "TestItem");

and then in a TestItem.php I have

```
use customiesdevs\customies\item\ItemComponents;
use customiesdevs\customies\item\ItemComponentsTrait;
use pocketmine\item\Item;
use customiesdevs\customies\item\CreativeInventoryInfo;

class TestItem extends Item implements ItemComponents {
	use ItemComponentsTrait;

	public function __construct() {
		$creativeInfo = new CreativeInventoryInfo(CreativeInventoryInfo::CATEGORY_EQUIPMENT, CreativeInventoryInfo::GROUP_SWORD);
		$this->initComponent("test", 1, $creativeInfo);
	}
}
```

I have no clue how to execute this "component" part. I'm probally just being dumb but I'm only a few weeks into learning pocketmine and I'm confused...

**Comment by jamie105** [comment]
I managed to get the blocks to work with textures, but I still don't know how to get the item components to work and add item textures

**Comment by unickorn** [comment, marked as answer]
You can add components using `$this->addComponent` after calling `$this->initComponent` in the constructor.

Check out the wiki! https://github.com/CustomiesDevs/Customies/wiki/Custom-Items

You can find a list of components over at https://bedrock.dev/docs/stable/Item.

**Reply by jamie105** [reply]
I understand that part, but I don't know how to "call" the component creation

**Reply by jamie105** [reply]
Also how to I specify a texture for the item?

**Reply by unickorn** [reply]
You don't call it! It's a constructor, it will get called when the item object is created.

**Reply by jamie105** [reply]
Thanks, after talking on discord I finally understand. I clearly have a shit ton to learn
